### PR TITLE
Bioscan tweaks

### PR DESCRIPTION
**Pull request by A-lexa**
:cl: LaKiller8
tweak: Bioscan now also shows a xeno location on the ship for the marines.
code: Made the Bioscan code more readable.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/cm_process.dm [hunk 1]
```diff
@@ -268,8 +269,10 @@ dat += " You failed to evacuate \the [MAIN_SHIP_NAME]"
 					if(0) //nullspace
 						if(M.loc && M.loc.z == 3) //in a closet or vent
 							numXenosShip++
+							xenoLocationsS += M.loc.loc.name
```
Comment (Rohesie) [inline code/game/gamemodes/cm_process.dm]: Make the `if(M.loc)` check run only once.
Or even better, use `get_area(M)`,which is cleaner and works a bit more reliably than checking the `locs` and accounting for `loc` inceptions.

**Comment by Rohesie** [comment]
`switch(M.z)` is not necessary if you run something like
```
var/area/A = get_area(M)
switch(A?.z)
```
That way you don't need to worry about locker or vent cases. The zero-case will be the actual nullspace, which shouldn't happen.

File: code/game/gamemodes/cm_process.dm [hunk 2]
```diff
@@ -298,38 +299,30 @@ dat += " You failed to evacuate \the [MAIN_SHIP_NAME]"
...
 			if(isXeno(M))
-				switch(M.z)
-					if(0) //nullspace
-						if(M.loc && M.loc.z == 3) //in a closet or vent
-							numXenosShip++
-						if(M.loc && M.loc.z == 1)
-							numXenosPlanet++
+				switch(A?.z)
 					if(1) //Planet.
 						if(istype(M, /mob/living/carbon/Xenomorph/Larva))
```
Comment (spookydonut) [inline code/game/gamemodes/cm_process.dm]:
```suggestion
						if(isXenoLarva(M))
```

File: code/game/gamemodes/cm_process.dm [hunk 3]
```diff
@@ -298,38 +299,30 @@ dat += " You failed to evacuate \the [MAIN_SHIP_NAME]"
...
 							numLarvaPlanet++
 						numXenosPlanet++
-						xenoLocationsP += M.loc.loc.name
+						xenoLocationsP += A
 					if(3) //On the ship.
 						if(istype(M, /mob/living/carbon/Xenomorph/Larva))
```
Comment (spookydonut) [inline code/game/gamemodes/cm_process.dm]:
```suggestion
						if(isXenoLarva(M))
```

File: code/game/gamemodes/cm_process.dm [hunk 4]
```diff
@@ -298,38 +299,30 @@ dat += " You failed to evacuate \the [MAIN_SHIP_NAME]"
...
-						if(M.loc && M.loc.z == 1)
-							numXenosPlanet++
+				switch(A?.z)
 					if(1) //Planet.
 						if(istype(M, /mob/living/carbon/Xenomorph/Larva))
 							numLarvaPlanet++
 						numXenosPlanet++
-						xenoLocationsP += M.loc.loc.name
+						xenoLocationsP += A
 					if(3) //On the ship.
```
Comment (spookydonut) [inline code/game/gamemodes/cm_process.dm]:
```suggestion
					if(MAIN_SHIP_Z_LEVEL) //On the ship.
```

File: code/game/gamemodes/cm_process.dm [hunk 5]
```diff
@@ -298,38 +299,30 @@ dat += " You failed to evacuate \the [MAIN_SHIP_NAME]"
...
-						if(M.loc && M.loc.z == 3) //in a closet or vent
-							numHostsShip++
-						if(M.loc && M.loc.z == 1)
-							numHostsPlanet++ 
+				switch(A?.z)
 					if(1) //Planet.
 						numHostsPlanet++ 
-						hostLocationsP += M.loc.loc.name
+						hostLocationsP += A
 					if(3) //On the ship.
```
Comment (spookydonut) [inline code/game/gamemodes/cm_process.dm]:
```suggestion
					if(MAIN_SHIP_Z_LEVEL) //On the ship.
```

File: code/game/gamemodes/cm_process.dm [hunk 6]
```diff
@@ -298,38 +299,30 @@ dat += " You failed to evacuate \the [MAIN_SHIP_NAME]"
...
+			var/area/A = get_area(M)
 			if(isXeno(M))
-				switch(M.z)
-					if(0) //nullspace
-						if(M.loc && M.loc.z == 3) //in a closet or vent
-							numXenosShip++
-						if(M.loc && M.loc.z == 1)
-							numXenosPlanet++
+				switch(A?.z)
 					if(1) //Planet.
```
Comment (spookydonut) [inline code/game/gamemodes/cm_process.dm]: add a define for like `#define SURFACE_Z_LEVEL 1`

**Pull request merged**